jekyll/_cci2/creating-orbs.md

```diff
@@ -0,0 +1,332 @@
+
+In development orbs, the string label given by the user has the following restrictions:
+
+* Up to 1023 non-whitespace characters
+
+Examples of valid development orb tags:
+
+* Valid:
+  "dev:mybranch"
+  "dev:2018_09_01"
+  "dev:1.2.3-rc1"
+  "dev:myinitials/mybranch"
+  "dev:myVERYIMPORTANTbranch"
+
+* Invalid
+  "dev: 1" (No spaces allowed)
+  "1.2.3-rc1" (No leading "dev:")
+
+In production orbs you must use the form ```X.Y.Z``` where ```X``` is a "major" version, ```Y``` is a "minor" version, and ```Z``` is a "patch" version. For example, 2.4.0 implies the major version 2, minor version 4, and the patch version of 0.
+
+While not strictly enforced, CircleCI recommends when versioning your production orbs you use the standard semantic versioning convention for deciding what major, minor, and patch should mean semantically:
+
+* MAJOR: when you make incompatible API changes
+* MINOR: when you add functionality in a backwards-compatible manner
+* PATCH: when you make backwards-compatible bug fixes
+
+#### Using Orbs in Orbs and Register-Time Resolution
+
+You may use an ```orbs``` stanza inside an orb, providing a way to pull in other orbs' elements into your orb. The scoping rules are the same as orbs in configuration, and you declare inline orbs in an orb.
+
+<aside class="notice">
+Because production orb releases are immutable, the system will resolve all orb dependencies at the time you register your orb rather than the time you run your build (eager resolution).
+
+For example, if you choose to publish orb ```foo/bar``` at version ```1.2.3```, and that orb you are publishing has, internally, an ```orbs``` stanza that imports another orb referenced as ```biz/baz@volatile```, the latest version at the time you want to perform this action is ```2.1.0```. At the time you register ```foo/bar@1.2.3``` the system will resolve ```biz/baz@volatile``` as the latest version and include its elements directly into the packaged version of ```foo/bar@1.2.3```.
+
+Now, if afterwards, ```biz/baz``` receives a new release that is now ```3.0.0```, anyone using ```foo/bar@1.2.3``` will not see the change in ```biz/baz@3.0.0``` until ```foo/bar``` is published at a higher version.
+</aside>
+
+#### Deleting Production Orbs
+
+In general, CircleCI prefers to never delete production orbs that were published as "Open" because it harms the reliability of the orb registry as a source of configuration and the trust of all orb users.
+
+If the case arises where you need to delete an orb for emergency reasons, please contact CircleCI (**Note:** If you are deleting because of a security concern, you must practice responsible disclosure: https://circleci.com/security/).
+
+#### Using the CLI to Create and Publish Orbs
+
+The ```circleci``` CLI has several handy commands for managing your orb publishing pipeline. The simplest way to learn the CLI is to install it and run ```circleci help```. Listed below are some of the most pertinent commands for publishing orbs:
```

Comment:
Is there an easy way to get to the CLI installation from here? (I'm assuming you have a few doc outline tricks up your sleeves, so if I'm asking silly questions please explain why so I know the answer for next time 😋 )

Reply:
Derp. Link is down below, wonderful.

Based on the provided link, it is a little unclear what token someone should be using with the CLI, or how to set it up.

Reply:
added link to cli doc